### ISSUE TITLE
bug(datepicker): navigating with ChromeVox keyboard shortcuts does not update selected date

### ISSUE THREAD

**Issue by zarend** [angular/components#23483]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "Datepicker action buttons" demo
3.  Activate "Open calendar" button
4. Linear navigate forward (Launcher + Right Arrow in ChromeOS)
5. press space
 
 
#### Expected Behavior
It behaves the same as pressing the right arrow key. It highlights the next day and selects the next day on closing.

#### Actual Behavior
Screenreader announces the next day, but does not select that day when closing

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox

##### Additional Notes
This is copied from an [internal bug report](http://b/195600355) found during a11y testing.

**Comment by zarend** [comment]
I can also repro on NVDA using the table specific shortcut keys – Ctrl + Alt + right arrow, Ctrl + Alt + left arrow, etc.

**Comment by zarend** [comment]
This works on other datepickers, such as the [Light Design React datepicker](https://react.lightningdesignsystem.com/components/date-pickers/), which also uses a table with role=grid. It also works on another internal datepicker. That means this is something we should be able to fix.

**Comment by Splaktar** [comment]
This may be related to [this newly opened issue](https://github.com/angular/components/issues/24230).
